src/db.c

```diff
@@ -456,6 +458,8 @@ long long emptyDb(int dbnum, int flags, void(callback)(dict*)) {
 
     if (dbnum == -1) flushSlaveKeysWithExpireList();
 
+    if (with_functions) functionsCtxClear(functionsCtxGetCurrent());
```

Comment:
maybe we should assert if this flag is passed and `dbnum` is not `-1`.

Reply:
Fixed